pkg/reconciler/mongodb/controller.go

```diff
@@ -14,4 +14,45 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package mongodb
+// Code generated by injection-gen. DO NOT EDIT.
+
+package mongodbsource
+
+import (
+	context "context"
+
+	mongodbsource "github.com/googleinterns/knative-source-mongodb/pkg/client/injection/informers/sources/v1alpha1/mongodbsource"
+	v1alpha1mongodbsource "github.com/googleinterns/knative-source-mongodb/pkg/client/injection/reconciler/sources/v1alpha1/mongodbsource"
+	secretinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/secret"
+	configmap "knative.dev/pkg/configmap"
+	controller "knative.dev/pkg/controller"
+	logging "knative.dev/pkg/logging"
+)
+
+// TODO: PLEASE COPY AND MODIFY THIS FILE AS A STARTING POINT
+
+// NewController creates a Reconciler for MongoDbSource and returns the result of NewImpl.
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+	logger := logging.FromContext(ctx)
+
+	mongodbsourceInformer := mongodbsource.Get(ctx)
+	secretInformer := secretinformer.Get(ctx)
+
+	// TODO: setup additional informers here.
+
+	r := &Reconciler{
+		secretLister: secretInformer.Lister(),
+	}
+	impl := v1alpha1mongodbsource.NewImpl(ctx, r)
+
+	logger.Info("Setting up event handlers.")
+
+	mongodbsourceInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+
```

Comment:
we also need to setup the secretInformer Event handler, so that whenever someone edits the secret, we are notified...

Something like https://github.com/vaikas/postgressource/blob/master/pkg/reconciler/postgressource/controller.go#L91

You can test it out creating a secret with an invalid connection string... you will see an error in the log when trying to connect to the database. After you edit the secret with kubectl edit using the correct credentials and save it, your reconciler will be called again, and successfully connect

Reply:
Thats weird I have in my file, not sure why it did not uploade

Reply:
Checked, need test